main/bluetooth/bluetooth.c

```diff
@@ -225,19 +227,56 @@ static esp_gatt_status_t telemetry_read(gatt_server_t *server, gatt_server_char_
     return ESP_GATT_OK;
 }
 
+static void bluetooth_default_device_name(rc_mode_e device_type, char* buf) {
+    char default_name[11];
```

Comment:
Pass the buffer size to this function and format everything into `buf` with one `snprintf()` call. e.g. `snprintf(buf, sizeof(buf), "%02X:%02X:%02X-%s", addr.addr[3], addr.addr[4], addr.addr[5], config_get_mode_name())`